### PR TITLE
Changes due to review of andsteing no resolved PR #609

### PR DESCRIPTION
**Pull request by bohnetbd**
a number of small changes due to comments of andsteing which have been not resolved in [PR #609](https://github.com/google/flax/pull/609).

### CODE REVIEW

File: examples/nlp_seq/train.py [hunk 1]
```diff
@@ -301,7 +294,7 @@ def main(argv):
   @jax.jit
```
Comment (andsteing) [inline examples/nlp_seq/train.py:294]: Let's change above comment to:

```
# jitting initialization avoids heap fragmentation
```

File: examples/nlp_seq/train.py [hunk 2]
```diff
@@ -20,26 +20,25 @@
 import tensorflow as tf
```
Comment (andsteing) [inline examples/nlp_seq/train.py:20]: BTW : why did you remove `perplexity` ?

File: examples/nlp_seq/models.py [hunk 3]
```diff
@@ -79,8 +78,7 @@ class AddPositionEmbs(nn.Module):
 
   @nn.compact
   def __call__(self,
-               inputs,
-               inputs_positions=None):
```
Comment (andsteing) [inline examples/nlp_seq/models.py:83]: wrt packed data : did it ever work?
(if yes: why removing this functionality?)

File: examples/nlp_seq/models.py [hunk 4]
```diff
@@ -205,14 +194,14 @@ class Transformer(nn.Module):
 
   @nn.compact
   def __call__(self,
-            inputs,
-            train=False,
-            dropout_rate=0.3):
+               *,
+               inputs,
```
Comment (andsteing) [inline examples/nlp_seq/models.py:198]: we usually have `inputs` (or `x`) as the first and only positional argument, i.e.

```
def __call__(self, inputs, *, train):
  ...
```

File: examples/nlp_seq/train.py [hunk 5]
```diff
@@ -301,7 +294,7 @@ def main(argv):
   @jax.jit
   def initialize_variables(init_rng):
     init_batch = jnp.ones((config.max_len, 1), jnp.float32)
-    init_variables = model.init(init_rng, init_batch)
+    init_variables = model.init(init_rng, inputs=init_batch, train=False)
```
Comment (andsteing) [inline examples/nlp_seq/train.py:297]: after my proposed change in `models.py`, you can revert here to

```
init_variables = model.init(init_rng, init_batch, train=True
```

(we usually say `train=True` for initialization)

**Review by andsteing — CHANGES_REQUESTED** [review]

File: examples/nlp_seq/models.py (hunk 3 above)
Comment (bohnetbd) [reply examples/nlp_seq/models.py:83]: I never tried and never used it basically - it got copied down from other transformer code.

File: examples/nlp_seq/models.py (hunk 4 above)
Comment (bohnetbd) [reply examples/nlp_seq/models.py:198]: ah. thanks! changed that.

File: examples/nlp_seq/train.py (hunk 2 above)
Comment (bohnetbd) [reply examples/nlp_seq/train.py:20]: It is unusual in part-of-speech tagging and many other tagging task. Further, I wanted the training loop more lean to make it more clear.

File: examples/nlp_seq/train.py (hunk 1 above)
Comment (bohnetbd) [reply examples/nlp_seq/train.py:294]: done

File: examples/nlp_seq/train.py (hunk 5 above)
Comment (bohnetbd) [reply examples/nlp_seq/train.py]: I changed this see, see new commit.
